pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/KitRequestShipping.java

```diff
@@ -996,19 +991,21 @@ public static String writeRequest(@NonNull String instanceId, @NonNull String dd
 
             DDPInstanceDto ddpInstanceDto =
                     new DDPInstanceDao().getDDPInstanceByInstanceId(Integer.valueOf(ddpInstance.getDdpInstanceId())).orElseThrow();
-
-            try {
-                UpsertPainlessFacade.of(DBConstants.DDP_KIT_REQUEST_ALIAS, kitRequestShipping, ddpInstanceDto,
-                        ESObjectConstants.DSM_KIT_REQUEST_ID, ESObjectConstants.DOC_ID,
-                        Exportable.getParticipantGuid(ddpParticipantId, ddpInstance.getParticipantIndexES()),
-                        new PutToNestedScriptBuilder()).export();
-            } catch (Exception e) {
-                //This error will trigger on studies with no participants, this skips
-                //the error log if that is the reason for the upsert failure.
-                if (StringUtils.isNotBlank((ddpInstance.getParticipantIndexES()))) {
-                    logger.error(String.format("Error inserting newly created kit request shipping with dsm kit request id: %s in "
-                            + "ElasticSearch", kitRequestShipping.getDsmKitRequestId()));
-                    e.printStackTrace();
+            // update ES only if it's a pepper study
+            if (StringUtils.isNotBlank(ddpInstance.getParticipantIndexES())) {
+                try {
+                    UpsertPainlessFacade.of(DBConstants.DDP_KIT_REQUEST_ALIAS, kitRequestShipping, ddpInstanceDto,
+                            ESObjectConstants.DSM_KIT_REQUEST_ID, ESObjectConstants.DOC_ID,
+                            Exportable.getParticipantGuid(ddpParticipantId, ddpInstance.getParticipantIndexES()),
+                            new PutToNestedScriptBuilder()).export();
+                } catch (Exception e) {
+                    //This error will trigger on studies with no participants, this skips
+                    //the error log if that is the reason for the upsert failure.
+                    if (StringUtils.isNotBlank((ddpInstance.getParticipantIndexES()))) {
```

Comment:
It seems like we did this check above to enter this code. Has anything changed since then?

Reply:
removed